rtl/eclrtl/rtlrecord.hpp

```diff
@@ -25,22 +25,189 @@
+
+    inline void onCreate(ICodeContext * _ctx) { ctx = _ctx; }
+
+protected:
+    ICodeContext * ctx;
+    unsigned activityId;
+
+};
+
+class COutputRowSerializer : implements IOutputRowSerializer, public CGlobalHelperClass
+{
+public:
+    inline COutputRowSerializer(unsigned _activityId) : CGlobalHelperClass(_activityId) { }
+    RTLIMPLEMENT_IINTERFACE
+
+    virtual void serialize(IRowSerializerTarget & out, const byte * self) = 0;
+};
+
+
+class COutputRowDeserializer : implements IOutputRowDeserializer, public RtlCInterface
+{
+public:
+    inline COutputRowDeserializer(unsigned _activityId) { activityId = _activityId; ctx = NULL; }
+    RTLIMPLEMENT_IINTERFACE
+
+    inline void onCreate(ICodeContext * _ctx) { ctx = _ctx; }
+
+    virtual size32_t deserialize(ARowBuilder & rowBuilder, IRowDeserializerSource & in) = 0;
+
+protected:
+    ICodeContext * ctx;
+    unsigned activityId;
+};
+
+
+class CSourceRowPrefetcher : implements ISourceRowPrefetcher, public RtlCInterface
+{
+public:
+    inline CSourceRowPrefetcher(unsigned _activityId) { activityId = _activityId; ctx = NULL; }
+    RTLIMPLEMENT_IINTERFACE
+
+    inline void onCreate(ICodeContext * _ctx) { ctx = _ctx; }
+
+    virtual void readAhead(IRowDeserializerSource & in) = 0;
+
+protected:
+    ICodeContext * ctx;
+    unsigned activityId;
+};
+
+
+class CFixedOutputRowSerializer : public COutputRowSerializer
+{
+public:
+    inline CFixedOutputRowSerializer(unsigned _activityId, unsigned _fixedSize) : COutputRowSerializer(_activityId) { fixedSize = _fixedSize; }
+
+    virtual void serialize(IRowSerializerTarget & out, const byte * self) { out.put(fixedSize, self); }
+
+protected:
+    size32_t fixedSize;
+};
+
+class CFixedOutputRowDeserializer : public COutputRowDeserializer
+{
+public:
+    inline CFixedOutputRowDeserializer(unsigned _activityId, unsigned _fixedSize) : COutputRowDeserializer(_activityId) { fixedSize = _fixedSize; }
+
+    virtual size32_t deserialize(ARowBuilder & rowBuilder, IRowDeserializerSource & in) { in.read(fixedSize, rowBuilder.getSelf()); return fixedSize; }
+
+protected:
+    size32_t fixedSize;
+};
+
+class CFixedSourceRowPrefetcher : public CSourceRowPrefetcher
+{
+public:
+    inline CFixedSourceRowPrefetcher(unsigned _activityId, unsigned _fixedSize) : CSourceRowPrefetcher(_activityId) { fixedSize = _fixedSize; }
+
+    virtual void readAhead(IRowDeserializerSource & in) { in.skip(fixedSize); }
+
+protected:
+    size32_t fixedSize;
+};
+
+//---------------------------------------------------------------------------
+
+class CXmlToRowTransformer : implements IXmlToRowTransformer, public CGlobalHelperClass
+{
+public:
+    inline CXmlToRowTransformer(unsigned _activityId) : CGlobalHelperClass(_activityId) {}
+    RTLIMPLEMENT_IINTERFACE
+
+};
+
+//---------------------------------------------------------------------------
+// Record size handling.
+
+class ThorHelper : public RtlCInterface
```

Comment:
Removed - nothing seems to have broken yet. It's not referenced from generated code?